### DISCUSSION TITLE
`ingest()` function for pulling data from `raw` to `mapped`

### DISCUSSION THREAD

**Discussion by jwildfire** [Gilead-Public/gsm#1832] [Ideas]
# Objective

Add a simple `Ingest()` step to our default gsm pipeline that creates a "basic" mapping layer.  `Ingest()` does the following. 
- Copies data sets from a `raw` data layer to a `mapped` data layer. 
    - Note: data set name are copied directly from the raw data layer
- Selects specified columns (drops all others) 
- Checks that column matches a specified type
- Renames specified columns

This ingest step will be driven by our standard `spec` object, generally speaking, `ingest()` will take `lRaw` and `spec` as inputs. lRaw can be a "connection" or a named list of dataFrames. 

After the ingest step, there will be an optional "Mapping" step where more complicated data mapping can take place using a standard gsm pipline (e.g. a yaml with `steps` passed to `RunWorkflows` that takes the the data created by `ingest()` as input. 

Highly related to [this discussion](https://github.com/Gilead-BioStats/gsm.endpoints/discussions/98).

**Comment by lauramaxwell** [comment]
I think this makes sense. So, the idea  that this would create the `Mapped_*` layer of data that will be a common set of tables used for all modules in the study. This is missing use of some common query features, like WHERE, GROUPBY, MIN, MAX, COUNT, etc. So, any filtering, grouping or aggregate functions (MIN, MAX, COUNT, etc) would be done in a second layer of mapping or within the module workflows themselves?

**Reply by jwildfire** [reply]
Yep - that pretty much nails it. Going to put together a quick prototype PR today/tomorrow.
